RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy

```diff
@@ -335,4 +336,106 @@ class BashConverter extends ConfigurationConverter {
+        xmlLines << "<configuration name='${getHeaderValue(header, "name", "")}'".toString()
+        xmlLines << "imports='" + getHeaderValue(header, "imports", "") + "'"
+        xmlLines << "description='" + getHeaderValue(header, "description", "") + "'"
+        xmlLines << "usedresourcessize='" + getHeaderValue(header, "usedresourcessize", "l") + "' >"
+
+        xmlLines << "  <availableAnalyses>"
+        getHeaderValues(header, "analysis", []).each {
+            String analysis ->
+                String[] split = analysis.split(StringConstants.COMMA)
+                String id = split[0]
+                String cfg = split[1]
+                String plg = split[2]
+                xmlLines << "    <analysis id='${id}' configuration='${cfg}' useplugin='${plg}' />".toString()
+        }
+        xmlLines << "  </availableAnalyses>"
+
+        // Convert the configuration values.
+        xmlLines << "  <configurationvalues>"
+
+        allLines[header.size()..-1].each {
+            String[] s = it.split("[=]")
+            if (!it) return;
+            if (!s) return;
+
+            if (isBashPipe(it)) {
+
+            } else if (isBashComment(it)) {
+                if (isBashComment(previousLine)) {
+                    xmlLines << ("""     ${it}""").toString()
+                } else {
+                    xmlLines << ("""<!-- ${it}""").toString()
+                }
+            } else {
+
+                if (isBashComment(previousLine)) {
+                    xmlLines[-1] += ("""-->""").toString()
+                }
+
+                if (s.size() == 2)
+                    xmlLines << ("""    <cvalue name='${s[0]}' value='${s[1]}' type='string' />""").toString()
+                else if (s.size() == 1)
+                    xmlLines << ("""    <cvalue name='${s[0]}' value='' type='string' />""").toString()
+                else
+                    xmlLines << ("""    <cvalue name='${s[0]}' value='${s[1..-1].join("=")}' type='string' />""").toString()
+            }
+            previousLine = it;
+        }
+        xmlLines << "  </configurationvalues>"
+        xmlLines << "</configuration>"
+
+        return xmlLines.join("\n")
+    }
+
+    public static String getHeaderValue(List<String> header, String id, String defaultValue) {
+        def found = header.find { String line -> line.startsWith("#${id}") }
+        if (!found) return defaultValue
+        return found.split(StringConstants.WHITESPACE)[1]
+    }
+
+    public static List<String> getHeaderValues(List<String> header, String id, List<String> defaultValue) {
+        def found = header.findAll { String line -> line.startsWith("#${id}") }
+        if (!found) return defaultValue
+        return found.collect { String entry -> entry.split(StringConstants.WHITESPACE)[1] }
+    }
+
+    public static boolean isBashComment(String it) {
+        it.trim().startsWith("#")
+    }
+
+    public static boolean isBashPipe(String it) {
+        it.trim().startsWith("###")
```

Comment:
That is a special form of code only used for one of our internal tools. I'll show you later. Normally it is not of use.